### PR TITLE
feat: add Twoslash to show TypeScript types

### PR DESCRIPTION
**Pull request by nakasyou**
I added [Twoslash](https://shiki.matsu.io/packages/vitepress#twoslash) to show TypeScript types.
Documentation users can check type definitions on the website.

スクリーンショット 2024-10-02 190235

Docs includes ambiguous codes, it was difficult to write correct TypeScript code. So I couldn't cover all of pages, but I think it has value.

### CODE REVIEW

File: package.json
Comment (yusukebe) [inline package.json:1]: You should add the `typescript` package.

**Comment by yusukebe** [comment]
Hi @nakasyou 

The feature was already suggested by @shinGangan #253

This looks good! But [as I commented](https://github.com/honojs/website/pull/253#issuecomment-1962365616), Hono uses lots of overloads, so some objects may be not good:

CleanShot 2024-10-03 at 10 56 05@2x

However, recently, we are writing JSDoc/TSDoc eagerly, so some, e.g. `notFound()` are great to see.

CleanShot 2024-10-03 at 10 57 12@2x

Basically I agree to introduce this feature.

Hi @shinGangan and others, what do you think of this?

**Comment by shinGangan** [comment]
Hi @yusukebe , I think it's a great idea, and I agree. It's okay to close #253 . ⭕️

Hi @nakasyou , thanks for creating this PR.

**Comment by nakasyou** [comment]
@yusukebe
> The feature was already suggested by @shinGangan https://github.com/honojs/website/pull/253

I'm sorry that I forgot to check duplicate PR. I'll close it.

> However, recently, we are writing JSDoc/TSDoc eagerly, so some

I agree with you.

**Comment by yusukebe** [comment]
Hi @nakasyou 

I think we don't have to close this PR. It's okay to close #253  and reopen this. Let's go with it.

**Comment by nakasyou** [comment]
Hi @yusukebe, I see.

Hi @shinGangan, thank you for proposing.

File: package.json
Comment (nakasyou) [reply package.json:1]: Fixed on https://github.com/honojs/website/pull/503/commits/6c149f9924b26a5a2ab9d1e418f99792aa881a23!

**Comment by yusukebe** [comment]
Hey @nakasyou 

Is this ready for review? I think it's okay to start withe minimal number of pages.

**Comment by nakasyou** [comment]
Hi @yusukebe, yes! It is ready to review. Covering all page is very hard, and I think we don't need covering all page in this PR.

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@nakasyou @shinGangan 

Looks good! Let's gooooo! Thank you for your contribution!

File: docs/api/request.md [hunk 1]
```diff
@@ -6,57 +6,73 @@ The `HonoRequest` is an object that can be taken from `c.req` which wraps a [Req
...
+```ts twoslash
+import { Hono } from 'hono'
+const app = new Hono()
+// ---cut---
 // Captured params
-app.get('/entry/:id', (c) => {
+app.get('/entry/:id', async (c) => {
   const id = c.req.param('id')
-  ...
+  //    ^?
```
Comment (yusukebe) [inline docs/api/request.md:16]: Is this necessary??

Comment (yusukebe) [reply docs/api/request.md:16]: @nakasyou 

Can you confirm this?

Comment (nakasyou) [reply docs/api/request.md:16]: I added it to make the return value easier to understand, but it might be unnecessary.

 I think that `c.req.queries()`, `c.req.parseBody()` and `c.req.header()` are easier to understand, but `c.req.param('id')` is not so. Should I remove this?

Comment (yusukebe) [reply docs/api/request.md:16]: No. If it's intended, it's okay! Thank you!
